src/main/java/org/jvnet/solaris/libzfs/LibZFS.java

```
@@ -39,17 +39,122 @@
+        Entry<String, String> myEntry;
+
+        System.out.println("[LIBZFS4J-DEBUG]: Setting '" +
+            libzfs4j_envvar_name + "' = '" + libzfs4j_usevar_value +
+            "' (envvar value was '" + libzfs4j_envvar_value + "')");
+        myEntry = new SimpleEntry<String, String>(libzfs4j_envvar_name, libzfs4j_usevar_value);
+        libzfs_features.add(myEntry);
+    }
+
+    void libzfs4j_init_features() {
+        String libzfs4j_envvar_value;
+        String libzfs4j_envvar_name;
+        String libzfs4j_usevar_value;
+        String libzfs4j_default_abi; /* Cache the default while we make decisions */
+
+        if (libzfs_features.size() > 0)
+            return; /* already inited */
+
+        libzfs4j_envvar_name = "LIBZFS4J_ABI";
+        libzfs4j_envvar_value = libzfs4j_get_extsetting(libzfs4j_envvar_name);
+        if (libzfs4j_envvar_value.equals("legacy") || libzfs4j_envvar_value.equals("openzfs")) {
+            /* Currently we recognize two values; later it may be more like openzfs-YYYY */
+            libzfs4j_default_abi = libzfs4j_envvar_value;
+        } else {
+            /* Detect presence of e.g. feature flags routines == openzfs */
+            libzfs4j_default_abi = "legacy";
+        }
+        libzfs4j_usevar_value = libzfs4j_default_abi;
+        libzfs4j_add_feature(libzfs4j_envvar_name, libzfs4j_usevar_value, libzfs4j_envvar_value);
+
+        libzfs4j_envvar_name = "LIBZFS4J_ABI_zfs_iter_snapshots";
+        libzfs4j_envvar_value = libzfs4j_get_extsetting(libzfs4j_envvar_name);
+        if (libzfs4j_envvar_value.equals("legacy") || libzfs4j_envvar_value.equals("3")) {
+            libzfs4j_usevar_value = "legacy";
+        } else {
+            if (libzfs4j_envvar_value.equals("openzfs") || libzfs4j_envvar_value.equals("4")) {
+                libzfs4j_usevar_value = "openzfs";
+            } else {
+                libzfs4j_usevar_value = libzfs4j_default_abi;
```

I would expect this method to fail in the case of the unsupported value